crates/kornia-core/src/tensor.rs

```diff
@@ -1304,4 +1380,103 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn display_2d() -> Result<(), TensorError> {
+        let data: [u8; 4] = [1, 2, 3, 4];
+        let t = Tensor::<u8, 2>::from_shape_slice([2, 2], &data, CpuAllocator)?;
+        let disp = t.to_string();
+        let lines = disp.lines().collect::<Vec<_>>();
+
+        #[rustfmt::skip]
+        assert_eq!(lines.as_slice(),
+        ["[[1,2],",
+         " [3,4]]"]);
+        Ok(())
+    }
+
+    #[test]
+    fn display_3d() -> Result<(), TensorError> {
+        let data: [u8; 12] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+        let t = Tensor::<u8, 3>::from_shape_slice([2, 3, 2], &data, CpuAllocator)?;
+        let disp = t.to_string();
+        let lines = disp.lines().collect::<Vec<_>>();
+
+        #[rustfmt::skip]
+        assert_eq!(lines.as_slice(),
+        ["[[[ 1, 2],",
+         "  [ 3, 4],",
+         "  [ 5, 6]],",
+         "",
+         " [[ 7, 8],",
+         "  [ 9,10],",
+         "  [11,12]]]"]);
+
+        Ok(())
+    }
+
+    #[test]
+    fn display_float() -> Result<(), TensorError> {
+        let data: [f32; 4] = [1.00001, 1.00009, 0.99991, 0.99999];
+        let t = Tensor::<f32, 2>::from_shape_slice([2, 2], &data, CpuAllocator)?;
+        let disp = t.to_string();
+        let lines = disp.lines().collect::<Vec<_>>();
+
+        #[rustfmt::skip]
+        assert_eq!(lines.as_slice(),
+        ["[[1.0000,1.0001],",
+         " [0.9999,1.0000]]"]);
+        Ok(())
+    }
+
+    #[test]
+    fn display_big_float() -> Result<(), TensorError> {
+        let data: [f32; 4] = [1000.00001, 1.00009, 0.99991, 0.99999];
+        let t = Tensor::<f32, 2>::from_shape_slice([2, 2], &data, CpuAllocator)?;
+        let disp = t.to_string();
+        let lines = disp.lines().collect::<Vec<_>>();
+
+        #[rustfmt::skip]
+        assert_eq!(lines.as_slice(),
+        ["[[1.0000e+03,1.0001e+00],",
+         " [9.9991e-01,9.9999e-01]]"]);
+        Ok(())
+    }
+
+    #[test]
+    fn display_big_tensor() -> Result<(), TensorError> {
+        let data: [u8; 1000] = [0; 1000];
+        let t = Tensor::<u8, 3>::from_shape_slice([10, 10, 10], &data, CpuAllocator)?;
+        let disp = t.to_string();
+        let lines = disp.lines().collect::<Vec<_>>();
+
+        #[rustfmt::skip]
+        assert_eq!(lines.as_slice(),
+        ["[[[0,0,0,...,0],",
```

Comment:
can be done later, but playing around with numpy/torch i've noticed that the magic number to truncate output is 10

![image](https://github.com/user-attachments/assets/de7b2502-a05a-4c4f-bb03-683ab5fd6362)
![image](https://github.com/user-attachments/assets/ee431080-cead-4017-8ad3-6e74ee0fa30c)